src/tests/Microsoft.PowerFx.Interpreter.Tests/UserInfoTests.cs

```diff
@@ -107,5 +171,98 @@ public void DisambiguationTest(string expr, object expected)
 
             Assert.Equal(expected, result.ToObject());
         }
+
+        // BasicUserInfo has same properties as UserInfo
+        [Fact]
+        public void Consistency()
+        {
+            Assert.Equal("User", UserInfo.ObjectName);
+
+            var methods = typeof(UserInfo).GetMethods(BindingFlags.Instance | BindingFlags.Public | BindingFlags.DeclaredOnly);
+            foreach (var method in methods)
+            {
+                var prop = typeof(BasicUserInfo).GetProperty(method.Name, BindingFlags.Public | BindingFlags.Instance);
+                Assert.NotNull(prop);
+            }
+        }
+
+        // Ensure that properties on the UserInfo object are not invoked until used. 
+        [Fact]
+        public async Task UserInfoLazy()
+        {
+            var userInfo = new MyUserInfo();
+
+            SymbolTable symbol = new SymbolTable();
+
+            var config = new PowerFxConfig();
+            config.SymbolTable.AddUserInfoObject(nameof(UserInfo.FullName));
+            
+            var engine = new RecalcEngine(config);
+
+            var rc = new RuntimeConfig();
+            rc.SetUserInfo(userInfo);
+
+            var check = engine.Check("User.FullName");
+            Assert.True(check.IsSuccess);
+            Assert.Equal(0, userInfo._counter);
+
+            var result = check.GetEvaluator().Eval(rc);
+            Assert.Equal(1, userInfo._counter);
+
+            Assert.Equal("full name", result.ToObject());
+        }
+
+        [Fact]
+        public async Task UserInfoCancel()
```

Comment:
this function and UserInfoLazy share similar code from 218 to 232. should we create share function?

Reply:
tempting, but there's too many variables passed in and out: userInfo, keys to AddUserInfoObject, engine/check, runtimeConfig, expression,